### PR TITLE
Return private key attributes on key pair generation.

### PR DESCRIPTION
**Pull request by rohitnarula7176**
## Proposed changes

Describe what this PR is trying to do.
Add ability to query creation date on RSA key pair.
## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m [hunk 1]
```diff
@@ -158,11 +158,9 @@ - (MSIDAssymetricKeyPair *)readKeyPairForAttributes:(MSIDAssymetricKeyLookupAttr
             return nil;
         }
         
-        NSDate *creationDate = [privateKeyDict objectForKey:(__bridge NSDate *)kSecAttrCreationDate];
-        
         MSIDAssymetricKeyPair *keypair = [[MSIDAssymetricKeyPair alloc] initWithPrivateKey:privateKeyRef
                                                                                  publicKey:publicKeyRef
-                                                                              creationDate:creationDate];
+                                                                            privateKeyDict:privateKeyDict];
```
Comment (oldalton) [inline IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m:163]: nit: should we call it something like "privateKeyAttributes"?

**Review by oldalton — APPROVED** [review]

File: IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m (hunk 1 above)
Comment (rohitnarula7176) [reply IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m:163]: When we generate keypair, we do not get back the private key dictionary so this param is set to the privateKey attributes but when you query private Key, we get the private key dictionary back so this param gets set to privateKeyDict. So I think we can keep the name as privateKeyDict. Let me know what you think.

Comment (oldalton) [reply IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m:163]: I don't have a strong preference.

Comment (rohitnarula7176) [reply IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m:163]: Perfect. So leaving it as is for now. :)

**Pull request merged**